src/main/java/no/unit/nva/model/Publication.java

```diff
@@ -178,57 +181,79 @@ public void setFileSet(FileSet fileSet) {
         this.fileSet = fileSet;
     }
 
+    public ResearchProject getProject() {
+        return project;
+    }
+
+    public void setProject(ResearchProject project) {
+        this.project = project;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof Publication)) {
```

Comment:
why is the null check removed?

Reply:
This is generated code, from what I can see, the null-check was only in place because the later call to getClass would throw an NPE. The generated code now obviates the need for the check because of the use of instanceof.